index.js

```diff
@@ -1,3 +1,10 @@
 'use strict';
 module.exports = typeof setImmediate === 'function' ? setImmediate :
-	function setImmediate(fn) { setTimeout(fn, 0); };
+  function setImmediate() {
+    var args = Array.prototype.slice.apply(arguments);
+    var fn = args.shift();
+    console.log('fn', fn, 'args', args);
+    setTimeout(function() {
```

Comment:
Yes, you can muck with the arguments to do that. I think it's still worth it to reduce the callstack on errrors and less overhead when used in a loop.